packages/next/server/app-render.tsx

```diff
@@ -1572,7 +1583,7 @@ export async function renderToHTMLOrFlight(
   )
 
   const requestStore = {
-    headers: new ReadonlyHeaders(headersWithoutFlight(req.headers)),
```

Comment:
Instead of creating a `lazy` method, IMHO it is possible to simply use a getter here on the `requestStore` object to achieve lazy evaluate.